sklearn/cross_validation.py

```diff
@@ -259,6 +259,10 @@ def __init__(self, n, n_folds, indices, shuffle, random_state):
         if not isinstance(shuffle, bool):
             raise TypeError("shuffle must be True or False;"
                             " got {0}".format(shuffle))
+
+        if random_state is not None and not shuffle:
+            warnings.warn('shuffle must be True if you need a non default random_state.')
```

Comment:
It should at least be used also in KFold. I'm not sure which other places. SGDClassifier, I guess?